### PR TITLE
#215 resolve aliases for group by exprs

### LINKED ISSUE #215: Group by Alias
**Describe the bug**
https://github.com/apache/arrow-datafusion/issues/110 suggests we can group by alias, however, I modify the existing test case `select_simple_aggregate_with_groupby_with_aliases` a bit, then it fails

**To Reproduce**
```
SELECT state AS a, MIN(age) AS b FROM person GROUP BY state
```
shows correct answer

```
SELECT state AS a, MIN(age) AS b FROM person GROUP BY a
```
raises error `Plan("Projection references non-aggregate values")`

**Expected behavior**
```
SELECT state AS a, MIN(age) AS b FROM person GROUP BY state

SELECT state AS a, MIN(age) AS b FROM person GROUP BY a
```
returns same answer

### PR DESCRIPTION
**Pull request by jychen7**
# Which issue does this PR close?

Closes #215 .

# What changes are included in this PR?

support alias in group by expression

# Are there any user-facing changes?

no

### CODE REVIEW

**Comment by jychen7** [comment]
cc @alamb 
it would be nice if you have time to help review

File: datafusion/src/sql/planner.rs [hunk 1]
```diff
@@ -691,14 +708,9 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
         input: &LogicalPlan,
         select_exprs: &[Expr],
         having_expr_opt: &Option<Expr>,
-        group_by: &[SQLExpr],
+        group_by_exprs: Vec<Expr>,
```
Comment (Dandandan) [inline datafusion/src/sql/planner.rs:711]: 👍

File: datafusion/src/sql/planner.rs [hunk 2]
```diff
@@ -2285,15 +2297,12 @@ mod tests {
...
-        let err = logical_plan(sql).expect_err("query should have failed");
-        assert_eq!(
-            format!(
-                "Plan(\"Invalid identifier \\\'x\\\' for schema {}\")",
-                PERSON_COLUMN_NAMES
-            ),
-            format!("{:?}", err)
+    fn select_simple_aggregate_with_groupby_can_use_alias() {
+        quick_test(
+            "SELECT state AS a, MIN(age) AS b FROM person GROUP BY a",
```
Comment (Dandandan) [inline datafusion/src/sql/planner.rs:2302]: 👍

**Review by Dandandan — APPROVED** [review]
Looks great to me @jychen7 !!

**Comment by alamb** [comment]
Thank you for the contribution @jychen7 !

**Pull request merged**